core/base/implicitTriangulation/ImplicitTriangulation.h

```diff
@@ -851,65 +853,90 @@ namespace ttk {
 
   public:
     inline SimplexId TTK_TRIANGULATION_INTERNAL(getVertexNeighborNumber)(
-      const SimplexId &vertexId) const override {
+      const SimplexId &vertexId) const final {
 
+      if(dimensionality_ == 3) {
+        const ttk::SimplexId z = vertexId / vshift_[1];
+        const auto idx2 = vertexId - (z * vshift_[1]);
+        const ttk::SimplexId y = idx2 / vshift_[0];
+        const ttk::SimplexId x = idx2 % vshift_[0];
+
+        int key = (x == 0              ? 1
+                   : x == nbvoxels_[0] ? 2
+                                       : 0)
+                  + (y == 0              ? 4
+                     : y == nbvoxels_[1] ? 8
+                                         : 0)
+                  + (z == 0              ? 16
+                     : z == nbvoxels_[2] ? 32
+                                         : 0);
+
+        constexpr std::array<ttk::SimplexId, 64> nNeighborsLUT{
+          14, 10, 10, 0, 10, 6, 8, 0, 10, 8, 6, 0, 0, 0, 0, 0,
+          10, 6,  8,  0, 8,  4, 7, 0, 6,  4, 4, 0, 0, 0, 0, 0,
+          10, 8,  6,  0, 6,  4, 4, 0, 8,  7, 4, 0, 0, 0, 0, 0,
+          0,  0,  0,  0, 0,  0, 0, 0, 0,  0, 0, 0, 0, 0, 0, 0};
+
+        return nNeighborsLUT[key];
+      } else {
 #ifndef TTK_ENABLE_KAMIKAZE
```

Comment:
that `ifndef` block should be moved before the test `if(dimensionality_ == 3) {` (l. 858)

Reply:
fixed